src/main/kotlin/org/radarbase/push/integration/common/auth/Oauth1Signing.kt

```diff
@@ -10,24 +10,25 @@ import java.security.GeneralSecurityException
 import java.util.*
 import javax.crypto.Mac
 import javax.crypto.spec.SecretKeySpec
+import kotlin.collections.HashMap
 
 class Oauth1Signing(
     val oauthKeys: OauthKeys,
     val nonce: String = UUID.randomUUID().toString(),
     val timestamp: Long = System.currentTimeMillis() / 1000L
 ) {
-
-    @Throws(IOException::class)
-    fun signRequest(request: Request): Request {
-
-        //Setup default parameters that will be sent with authorization header
-        val parameters = hashMapOf(
+    fun getParams(): HashMap<String, String>
+         = hashMapOf(
             OAUTH_CONSUMER_KEY to oauthKeys.consumerKey,
             OAUTH_NONCE to nonce,
             OAUTH_SIGNATURE_METHOD to OAUTH_SIGNATURE_METHOD_VALUE,
             OAUTH_TIMESTAMP to timestamp.toString(),
             OAUTH_VERSION to OAUTH_VERSION_VALUE
         )
+
+    @Throws(IOException::class)
+    fun signRequest(request: Request, parameters: HashMap<String, String>, signature: String): Request {
```

Comment:
I think you can move the params out of this class (as they are not required here anymore) and pass them as class vars (removing timestamp and nonce).